### DISCUSSION TITLE
Option for Pinned Tabs to appear in SINGLE Workspaces, or ALL Workspaces as now.

### DISCUSSION THREAD

**Discussion by mr-cheffy** [zen-browser/desktop#875] [Ideas]
### What feature would you like to see?

Afaict, atm all Pinned Tabs appear across ALL Workspaces. 

If possible, pls could we have an **option to have tabs be pinned per Workspace, rather than all Workspaces?** 

### What platform are you seeing the problem on?

Linux

### Additional context

I don't know if this is helpful or not wrt design for this feature request, but the fantastic Add-On `Sidebery` provides this capability [but atm `Sidebery` is not terribly compatible with Zen, so i only use it in `Firefox`, `Firefox Nightly`, & `Floorp`]. 
 
 ~ _Created automatically from issue: https://github.com/zen-browser/desktop/issues/238_

**Comment by ChristinWhite** [comment]
I don't think Sidebery supports it, but I'd love to have the option to set pinned tabs to be global or local to the workspace selectively rather than a global setting.

**Comment by keifererikson** [comment]
This is the main thing holding me back from using Zen a lot more. In Arc, you have favourites, and pinned tabs/folders. I love having different pinned/favourites depending on workspace. It allows for a lot smoother workflow with different projects, or school classes, or jobs. I'm hoping this is a feature that can be implemented in the future.

**Reply by lucianoratamero** [reply]
same here. I usually don't even use bookmarks, since they're not that straightforward to find and manage.
the only thing keeping me in arc is having pinned folders and tabs on specific workspaces.

a way to sync them between computers would also be high in priority for my use cases, but they're not a deal breaker for me, since I can manually put them in bookmarks if needed. in fact, my ideal scenario would be for pinned tabs and folders to **be**
bookmarks, but I'm not sure how feasible that would be.

**Reply by mark-rifkin** [reply]
Pinned folders and tabs are also the main feature keeping me in Arc. If implemented in Zen I would love to have "favorites" with just the favicon (which I use for email and calendar) as well as normal pinned tabs per workspace. In arc the favorites are per profile, not per workspace, though it wouldn't be too annoying to have them per workspace. It's great to be able easily open sites relevant to each project with much less friction than standard bookmarks. 

The other related feature would be tab clearing of non-pinned tabs, I haven't found something that does this well in Firefox in the past.

**Comment by jandroav** [comment]
The only workaround I can see is having a 1-1 relation between profiles, workspaces and pinned tabs but it can only be uses to have 1 workspaces on 1 profile.

**Comment by Alexcoder5** [comment]
I use Sidebery with Zen, but had to disable Zen tab panel. If you are interested, I can share my CSS.

**Comment by little-rocket100** [comment]
This is the only reason that Zen isn't working for me and I've gone back to Arc. I spent all day yesterday using various extensions and tools to try to mimic the same functionality as arc ie to show different pinned tabs and bookmarks for each workspace, but I couldn't make it happen. I will come back to Zen when this is in place.

**Comment by alizehctl** [comment]
Was about to make an issue for exactly this feature. To me, having pinned tabs persist across workspaces doesn't make sense.

For example, I keep my workspace chat, email, time tracker, and task list pinned in my browser when I'm working. When I switch to my "Personal" profile in Zen, I still feel like I'm stuck in work mode because all my work tabs are still pinned.

**Comment by rawkode** [comment]
I think the problem here is the Firefox implementation of pinned tabs. However, I think there's an alternative solution.

I'll speak for myself and not everybody else, but the reason I use pinned tabs is to stop the tab from being closed. Ideally it'd lock it to a specific URL prefix, but that doesn't exist atm.

Perhaps Zen could implement "Lock Tabs" which provide the same behavior with workspace isolation, rather than using Firefox's global pinned tab feature?

**Comment by 0x2f0** [comment]
I also think it's a good idea to have tabs that can be pinned for a local workspace and globally as well. There are certain times where both may be useful. I love how arc does it. It would be cherry on top to have that feature in zen as well.

**Comment by artur-rios** [comment]
This is a main feature to me, and is holding me for using zen a lot more

**Comment by RustedAperture** [comment]
I came here to look specifically for this, I would prefer if the pinned tabs were specific to the workspace they are in. Right now I use two profiles in two windows. one for work and the other for personal.

**Comment by kylescudder** [comment]
This appears to be working for me now as of 1.0.1-a.2.

**Comment by Hikiru** [comment]
We need separate pinned tabs like Arc. I have universal pinned tabs that I use everywhere then workspace specific pinned tabs below.

**Comment by jasiralavibiztripz** [comment]
I think it would be good to have two different functions:
1. Pinned tabs: Tabs are pinned on the current workspace only
2. Starred tabs: Tabs appear across all workspaces.

Probably position/order them as: Starred, Pinned, then Other tabs.

**Reply by xanthos84** [reply]
this!

sad this discussion was already closed.

**Reply by jasiralavibiztripz** [reply]
add as a new feature request?

**Reply by thebiblelover7** [reply]
@jasiralavibiztripz Yes, that's a good idea

**Reply by jasiralavibiztripz** [reply]
Done! https://github.com/zen-browser/desktop/issues/2525
